eth_pydantic_types/hex.py

```diff
@@ -115,13 +170,38 @@ def from_bytes(cls, data: bytes) -> "HexStr":
         return HexStr(value)
 
 
-def validate_hex_str(value: str) -> str:
-    hex_value = (value[2:] if value.startswith("0x") else value).lower()
-    if set(hex_value) - set("1234567890abcdef"):
-        raise HexValueError(value)
+class BoundHexStr(BaseHexStr):
+    """A hex string value, typically from a hash, that is required to be a specific size."""
+
+    size: ClassVar[int] = 32
+    calculate_schema: ClassVar[bool] = True
+
+    @classmethod
+    def __get_pydantic_core_schema__(cls, value, handler=None) -> "CoreSchema":
+        str_size = cls.size * 2 + 2
+        return with_info_before_validator_function(
+            cls.__eth_pydantic_validate__,
+            str_schema(max_length=str_size, min_length=str_size),
+        )
+
+    @classmethod
+    def __eth_pydantic_validate__(cls, value: Any, info: Optional[ValidationInfo] = None) -> str:
+        hex_str = cls.validate_hex(value)
+        hex_value = hex_str[2:] if hex_str.startswith("0x") else hex_str
+        sized_value = cls.validate_size(hex_value)
+        return cls(f"0x{sized_value}")
+
+    @classmethod
+    def validate_size(cls, value: str) -> str:
+        if cls.calculate_schema:
```

Comment:
question: instead of using `calculate_schema`, what about overriding `validate_size()` in Address?

Reply:
addressed by overriding not the full `validate_size()` call but extracted the contents of this conditional to its own call and overrode that